### PR TITLE
[IMP] pos: New FloorScreen kanban view and more

### PR DESCRIPTION
**Pull request by adgu-odoo**
Currently, there is only 1 plan view that is unreadable on mobile and sometimes also complicated in some cases.
This commit  adds a new "Kanban" view used by default on mobile and available in desktop too. This commit also changes the edit bar of the FloorScreen: the design is changed and some features are added like the multiple selection of the tables or the adding and removing of floors directly from the frontend. It also changes the navbar of the PoS, moving the buttons in a navigation menu accessible thanks to a button.

Related Enterprise PR: odoo/enterprise#38699

Task-id: 3212032

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/point_of_sale/static/src/scss/pos.scss [hunk 1]
```diff
@@ -3759,3 +3723,36 @@ td {
...
+                
+                span, i {
+                    color: black;
+                    inline-size: 196px;
+                    overflow-wrap: break-word;
+                }
+            }
+		}
+	}
+}
```
Comment (trj-odoo) [inline addons/point_of_sale/static/src/scss/pos.scss:3758]: t'es sérieux eof ?

File: addons/point_of_sale/static/src/scss/pos.scss [hunk 2]
```diff
@@ -3759,3 +3723,36 @@ td {
...
+                padding: 10px;
+                
+                span, i {
+                    color: black;
+                    inline-size: 196px;
+                    overflow-wrap: break-word;
+                }
+            }
+		}
+	}
```
Comment (trj-odoo) [inline addons/point_of_sale/static/src/scss/pos.scss:3757]: Pourquoi cela part-il en cacahuètes ? (indentation)

File: addons/pos_restaurant/models/pos_restaurant.py [hunk 3]
```diff
@@ -41,6 +41,40 @@ def write(self, vals):
                         'Open session: %s' % (' '.join(config.mapped('name')),))
         return super(RestaurantFloor, self).write(vals)
 
+    def rename_floor(self, new_name):
+        """ create or modify a floor from the point of sale UI.
+            floor contains the floor's fields. If it contains an
+            id, it will modify the existing floor. It then
+            returns the id of the floor.
```
Comment (trj-odoo) [inline addons/pos_restaurant/models/pos_restaurant.py:48]: bullshit (with love), you are just modifying the name

File: addons/pos_restaurant/static/src/app/floor_screen/floor_screen.js [hunk 4]
```diff
@@ -390,24 +563,46 @@ export class FloorScreen extends Component {
...
         // Since we wanted to disable the selected table after deletion, we should be
         //   setting the selectedTableId to null. However, we only do this if nothing
         //   else is selected during the rpc call.
-        if (this.state.selectedTableId === originalSelectedTableId) {
-            this.state.selectedTableId = null;
+        const equalsCheck = (a, b) => {
+            return JSON.stringify(a) === JSON.stringify(b);
+        }
+        if (equalsCheck(this.state.selectedTableIds, originalSelectedTableIds)) {
+            this.state.selectedTableIds.splice(0, this.state.selectedTableIds.length);
```
Comment (trj-odoo) [inline addons/pos_restaurant/static/src/app/floor_screen/floor_screen.js:604]:
```suggestion
            this.state.selectedTableIds = [];```

File: addons/pos_restaurant/static/src/app/floor_screen/floor_screen.js [hunk 5]
```diff
@@ -390,24 +563,46 @@ export class FloorScreen extends Component {
...
         //   before the await call can be different after the finishing the await call.
         // Since we wanted to disable the selected table after deletion, we should be
         //   setting the selectedTableId to null. However, we only do this if nothing
         //   else is selected during the rpc call.
-        if (this.state.selectedTableId === originalSelectedTableId) {
-            this.state.selectedTableId = null;
+        const equalsCheck = (a, b) => {
+            return JSON.stringify(a) === JSON.stringify(b);
+        }
+        if (equalsCheck(this.state.selectedTableIds, originalSelectedTableIds)) {
```
Comment (trj-odoo) [inline addons/pos_restaurant/static/src/app/floor_screen/floor_screen.js:603]: careful if the order is important

File: addons/pos_restaurant/static/src/app/floor_screen/table.js [hunk 6]
```diff
@@ -24,16 +24,41 @@ export class Table extends Component {
...
-            background: ${table.color || "rgb(53, 211, 116)"};
-            font-size: ${table.height >= 150 && table.width >= 150 ? 32 : 16}px;
-        `;
+
+        if (this.env.pos.floorPlanStyle == "kanban") {
+            const floor = table.floor;
+            const index = floor.tables.indexOf(table);
+
+            const minWidth = 100 + 20;
+
```
Comment (trj-odoo) [inline addons/pos_restaurant/static/src/app/floor_screen/table.js:33]: random space

File: addons/pos_restaurant/static/src/app/floor_screen/table.js [hunk 7]
```diff
@@ -24,16 +24,41 @@ export class Table extends Component {
...
+            let nbrHorizontal = Math.floor(window.innerWidth / minWidth);
+
+            let widthTable = (window.innerWidth - nbrHorizontal*10) / nbrHorizontal;
+            let position_h = widthTable * (index % nbrHorizontal) + 5 + (index % nbrHorizontal) * 10;
+            let position_v = widthTable * Math.floor(index / nbrHorizontal) + 5 + Math.floor(index / nbrHorizontal) * 10
```
Comment (trj-odoo) [inline addons/pos_restaurant/static/src/app/floor_screen/table.js:38]: j'crois vraiment tu peux faire des consts

File: addons/pos_restaurant/static/src/app/navbar/navbar.js [hunk 8]
```diff
@@ -25,4 +25,14 @@ patch(Navbar.prototype, "pos_restaurant.Navbar", {
             !this.env.pos.table
         ));
     },
+    onSwitchButtonClick() {
+        if (this.env.pos.floorPlanStyle == "kanban") {
+            this.env.pos.floorPlanStyle = "default";
+        } else {
+            this.env.pos.floorPlanStyle = "kanban";
```
Comment (trj-odoo) [inline addons/pos_restaurant/static/src/app/navbar/navbar.js:32]:
```suggestion
        this.env.pos.floorPlanStyle = this.env.pos.floorPlanStyle == "kanban" ?  "default" : "kanban";
```

File: addons/pos_restaurant/static/src/app/navbar/navbar.xml [hunk 9]
```diff
@@ -2,6 +2,14 @@
 <templates id="template" xml:space="preserve">
 
     <t t-name="pos_restaurant.Navbar" t-inherit="point_of_sale.Navbar" t-inherit-mode="extension">
+        <xpath expr="//div[hasclass('end-nav-button')]" position="before">
+            <li t-if="pos.mainScreen.component.name == 'FloorScreen'" class="menu-item navbar-button edit-button" t-on-click="toggleEditMode">
+                <span>Edit Plan</span>
+            </li>
+            <li t-if="pos.mainScreen.component.name == 'FloorScreen'" class="menu-item navbar-button" t-on-click="onSwitchButtonClick">
+                <span>Switch Floor Plan</span>
```
[the remark this answers is not in the document]
Comment (trj-odoo) [reply addons/pos_restaurant/static/src/app/navbar/navbar.xml:10]: C'est Alain qui l'a dit

File: addons/pos_restaurant/static/src/js/Chrome.js [hunk 10]
```diff
@@ -10,7 +10,7 @@ patch(Chrome.prototype, "pos_restaurant.Chrome", {
      * `FloorScreen` is the start screen if there are floors.
      */
     get startScreen() {
-        if (this.env.pos.config.iface_floorplan) {
+        if (this.env.pos.config.iface_floorplan || this.env.pos.config.is_table_management) {
```
Comment (trj-odoo) [inline addons/pos_restaurant/static/src/js/Chrome.js:13]: Don't forget to remove this one because is_table_management will replae iface_floorplan

**Review by trj-odoo — CHANGES_REQUESTED** [review]
This is my last and final review at Odoo, you did a great job. Well done. Keep this up and you'll become the very best ~~pokemon trainer~~ at what you do.

It was nice reviewing this alongside you. 
Enjoy the last work i'm giving you

File: addons/pos_restaurant/static/tests/tours/FloorScreen.tour.js [hunk 11]
```diff
@@ -24,41 +24,36 @@ FloorScreen.check.hasTable("1");
...
 FloorScreen.do.clickAddTable();
 FloorScreen.check.selectedTableIs("1");
 FloorScreen.do.clickRename();
 TextInputPopup.check.isShown();
 TextInputPopup.do.inputText("100");
 TextInputPopup.do.clickConfirm();
+FloorScreen.do.clickTable("100");
 FloorScreen.check.selectedTableIs("100");
 
 // test duplicate table
```
Comment (rhe-odoo) [inline addons/pos_restaurant/static/tests/tours/FloorScreen.tour.js:43]: Can you test duplicate on multiple tables? (=> control click)

File: addons/pos_restaurant/static/src/js/models.js [hunk 12]
```diff
@@ -16,6 +16,8 @@ patch(PosGlobalState.prototype, "pos_restaurant.PosGlobalState", {
         this.orderToTransfer = null; // table transfer feature
         this.transferredOrdersSet = new Set(); // used to know which orders has been transferred but not sent to the back end yet
         this.printers_category_ids_set = new Set();
+        this.floorPlanStyle = "";
```
Comment (rhe-odoo) [inline addons/pos_restaurant/static/src/js/models.js:19]: Shouldn't be it default?

**Comment by rhe-odoo** [comment]
@robodoo r+

**Pull request merged**